keps/sig-release/20190316-rebase-images-to-distroless.md

```diff
@@ -0,0 +1,167 @@
+
+Currently, there are **three** different workflows to build the core master images and which workflow to use  is determined by each **cloud provider**.
+
+####[Bash Release](https://github.com/kubernetes/kubernetes/blob/caf9d94d697ce327e0c1c3dee71a1f06a6fc918e/build/root/Makefile#L419)
+
+Run `make quick-release` under kubernetes repo. This approach is most commonly used and it uses the bash scripts (See [build/release.sh](https://github.com/kubernetes/kubernetes/blob/caf9d94d697ce327e0c1c3dee71a1f06a6fc918e/build/release.sh#L36) for details) to build the images. In this workflow, the base image is specified in the [`build/common.sh`](https://github.com/kubernetes/kubernetes/blob/f26048ecb1c7b6fb67c2e7c7c96070d7a1743d86/build/common.sh#L96).
+
+####[Bazel Release](https://github.com/kubernetes/kubernetes/blob/caf9d94d697ce327e0c1c3dee71a1f06a6fc918e/build/root/Makefile#L604)
+
+Run `make bazel-release` under kubernetes repo. This approach uses bazel to build the image artifact based on this [BUILD](https://github.com/kubernetes/kubernetes/blob/caf9d94d697ce327e0c1c3dee71a1f06a6fc918e/build/release-tars/BUILD) rule (More details in [bazel.bzl/release-filegroup](https://github.com/kubernetes/repo-infra/blob/4528e18f5d62a2a5172f76d198738d85d4d04734/defs/build.bzl#L134)) . In this workflow, the base image is specified in the [build/BUILD](https://github.com/kubernetes/kubernetes/blob/3fd6f97f55d51c01df3c01c7ffbb2834c25d9900/build/BUILD#L31) rule.
+
+####[Test Release](https://github.com/kubernetes/test-infra/tree/master/kubetest)
+
+Run `kubetest` or `hack/e2e`. See details in the [test-infra repo](https://github.com/kubernetes/test-infra/tree/master/kubetest). This approach is recommended for development testing and is broadly used by contributors. However, this approach is under a test env and it uses different config than the two official workflows as described above. In this workflow, the base image is specified to use k8s.gcr.io/pause:3.1.
+
+###Solution
+
+This KEP is expected to rebase images for all three workflows. This requires each cloud provider team to be involved in the manifest updates and release workflow testing part (See the graph below). Before we switch the base images to `distroless/static`, each cloud provider team should make sure their manifest config is updated so that the **command doesn’t require shell to run the executable binaries and no log redirection is involved in the command**. Otherwise rebasing images to distroless will **break** the core containers running in the cluster master VMs. The test release should also be updated to `distroless/static` so as we can guarantee further changes wouldn’t be able to add unexpected dependencies (otherwise, they will fail the e2e tests in the github prow test stage).
+
+![Rebase Core Master](RebaseCoreMaster.png?raw=true "Rebase Core Master")
```

Comment:
Just curious, how do you review the Markdown PR? I added a graph here and it seems not showing up in the .md nor the original png file.

Reply:
It shows up. Just navigate to your brach on your personal fork. You can also get there directly by clicking "view file" from this PR: https://github.com/kubernetes/enhancements/blob/466f031aa70f69353abbf64ab828eb7a6da6e429/keps/sig-release/20190316-rebase-images-to-distroless.md#distroless-and-previous-work

Reply:
Ah, I see. Thanks.